homeassistant/components/statistics/sensor.py

```diff
@@ -281,22 +281,22 @@ def _next_to_purge_timestamp(self):
 
         # If max_age is set, ensure to update again after the defined interval.
         next_to_purge_timestamp = self._next_to_purge_timestamp()
-        if next_to_purge_timestamp:
+        if (
+            next_to_purge_timestamp is not None
+            and (next_to_purge_timestamp - self._next_purge).total_seconds() > 1
+        ):
+            self._next_purge = next_to_purge_timestamp
             _LOGGER.debug(
-                "%s: scheduling update at %s", self.entity_id, next_to_purge_timestamp
+                "%s: scheduling update at %s", self.entity_id, self._next_purge
```

Comment:
This is a bit misleading, because `async_track_point_in_utc_time` is actually called with `next_to_purge_timestamp` below, so I would not change this line.

Reply:
I'll revert it